src/sass/parse.js

```diff
@@ -4177,34 +4180,68 @@ function getSC() {
 }
 
 /**
- * Check if token is part of a hexadecimal number (e.g. `#fff`) inside
- *      a simple selector
+ * Check if token is part of a hexadecimal number (e.g. `#fff`) inside a simple
+ * selector
  * @param {number} i Token's index number
  * @return {number}
  */
 function checkShash(i) {
-  var l;
+  const start = i;
+  let l;
 
-  if (i >= tokensLength || tokens[i].type !== TokenType.NumberSign) return 0;
+  if (i >= tokensLength) return 0;
+
+  if (tokens[i].type === TokenType.NumberSign) {
+    if (checkInterpolation(i)) return 0;
+    i++;
+  } else return 0;
+
+  while (i < tokensLength) {
+    if (l = checkIdentOrInterpolation(i) || checkNumber(i)) i += l;
+    else if (tokens[i].type === TokenType.HyphenMinus) i += 1;
```

Comment:
I've added `checkPartialIdent()` function (1) which should check if a string is `nmchar` according to the grammar (2). The function also has a getter which can be used in `getShash()`.

Do we want to check for that "partial ident" in every case we meet interpolation or are there any exceptions? If there are none, we could add `checkPartialIdent` inside `checkIdentOrInterpolation`.

(1): https://github.com/tonyganch/gonzales-pe/blob/dev/src/scss/parse.js#L1739
(2): https://www.w3.org/TR/CSS21/grammar.html